_dev/js/product.js

```diff
@@ -29,15 +29,46 @@ import ProductSelect from './components/product-select';
 $(document).ready(() => {
   function coverImage() {
     const productCover = $(prestashop.themeSelectors.product.cover);
+    const modalProductCover = $(prestashop.themeSelectors.product.modalProductCover);
     let thumbSelected = $(prestashop.themeSelectors.product.selected);
 
     const swipe = (selectedThumb, thumbParent) => {
       const newSelectedThumb = thumbParent.find(prestashop.themeSelectors.product.thumb);
 
-      $(prestashop.themeSelectors.product.modalProductCover).attr('src', newSelectedThumb.data('image-large-src'));
+      // Swap active classes on thumbnail
       selectedThumb.removeClass('selected');
       newSelectedThumb.addClass('selected');
+
+      // Update sources of both cover and modal cover
+      modalProductCover.prop('src', newSelectedThumb.data('image-large-src'));
       productCover.prop('src', newSelectedThumb.data('image-medium-src'));
+
+      // Get data from thumbnail and update cover src, alt and title
+      productCover.attr('title', newSelectedThumb.attr('title'));
+      modalProductCover.attr('title', newSelectedThumb.attr('title'));
+      productCover.attr('alt', newSelectedThumb.attr('alt'));
+      modalProductCover.attr('alt', newSelectedThumb.attr('alt'));
+
+      // Get data from thumbnail and update cover sources
+      const sourcesMedium = newSelectedThumb.data('image-medium-sources');
+      const sourcesLarge = newSelectedThumb.data('image-large-sources');
+      const productCoverWebp = productCover.siblings('source[type="image/webp"]');
+      const productCoverAvif = productCover.siblings('source[type="image/avif"]');
+      const modalProductCoverWebp = modalProductCover.siblings('source[type="image/webp"]');
+      const modalProductCoverAvif = modalProductCover.siblings('source[type="image/avif"]');
+
+      if (sourcesMedium !== undefined && sourcesMedium.webp !== undefined && productCoverWebp.length) {
```

Comment:
You are using the same code in both files, maybe it could be a separate component where you provide `sources` and information about the cover?

Reply:
Done ✅